pootle/static/js/editor/app.js

```diff
@@ -124,40 +124,54 @@ PTL.editor = {
     this.$msgOverlay = $('#js-editor-msg-overlay');
     this.$navNext = $('#js-nav-next');
     this.$navPrev = $('#js-nav-prev');
+    this.unitPositionEl = q('.js-unit-position');
     this.unitCountEl = q('.js-unit-count');
-    this.unitIndexEl = q('.js-unit-index');
-    this.offsetRequested = 0;
 
     /* Initialize variables */
-    this.units = new UnitSet([], {
-      chunkSize: this.settings.chunkSize,
-    });
-    this.editorRow = null;
 
     this.filter = 'all';
     this.checks = [];
     this.sortBy = 'default';
     this.modifiedSince = null;
     this.user = null;
-    this.ctxGap = 0;
-    this.ctxQty = parseInt(cookie('ctxQty'), 10) || 1;
     this.preventNavigation = false;
-
+    this.units = new UnitList();
     this.isUnitDirty = false;
+    this.editorIsClosed = false;
+    this.contextTimer = null;
+    this.lastFilter = null;
+
+    this.units.onUnitChange = (uid) => {
+      this.handleUnitChange(uid);
+    };
+
+    this.units.onViewRowsChange = (uid) => {
+      if (uid === this.units.uid) {
+        this.handleViewRowsChange();
+      }
+    };
+
+    this.units.onContextData = (uid) => {
+      if (uid === this.units.uid) {
```

Comment:
We could incorporate this uid check into UnitsList itself if we were sure that the higher level code doesn't need to get the callback at all (for e.g. doing some cleanups). So what I decided to do for now is to run the callback anyway and let the higher level code deal with the outdated callbacks. But we may want to move these checks into UnitList in the future if this results in a cleaner app code.

I agree that passing the data in a callback might be a cleaner approach.